apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -22,6 +22,520 @@ import (
+const CIPHER5 TLSCipherType = "ECDHE-ECDSA-AES128-SHA"
+const CIPHER6 TLSCipherType = "ECDHE-RSA-AES128-SHA"
+const CIPHER7 TLSCipherType = "AES128-GCM-SHA256"
+const CIPHER8 TLSCipherType = "AES128-SHA"
+const CIPHER9 TLSCipherType = "ECDHE-ECDSA-AES256-GCM-SHA384"
+const CIPHER10 TLSCipherType = "ECDHE-RSA-AES256-GCM-SHA384"
+const CIPHER11 TLSCipherType = "ECDHE-ECDSA-AES256-SHA"
+const CIPHER12 TLSCipherType = "ECDHE-RSA-AES256-SHA"
+const CIPHER13 TLSCipherType = "AES256-GCM-SHA384"
+const CIPHER14 TLSCipherType = "AES256-SHA"
+
+// EnvoyTLS describes tls parameters for Envoy listneners.
+type EnvoyTLS struct {
+	// MinimumProtocolVersion is the minimum TLS version this vhost should
+	// negotiate. Valid options are `1.2` (default) and `1.3`.
+	// +kubebuilder:default="1.2"
+	// +kubebuilder:validation:Enum="1.2";"1.3"
+	MinimumProtocolVersion string `json:"minimumProtocolVersion"`
+
+	// CipherSuites defines the TLS ciphers to be supported by Envoy TLS
+	// listeners when negotiating TLS 1.2. Ciphers are validated against the
+	// set that Envoy supports by default. This parameter should only be used
+	// by advanced users. Note that these will be ignored when TLS 1.3 is in
+	// use.
+	//
+	//See: https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto#extensions-transport-sockets-tls-v3-tlsparameters
+	// Note: This list is a superset of what is valid for stock Envoy builds and those using BoringSSL FIPS.
+	//
+	// +kubebuilder:default="[ECDHE-ECDSA-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]";"[ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POLY1305]";"ECDHE-ECDSA-AES256-GCM-SHA384";"ECDHE-RSA-AES256-GCM-SHA384"
+	CipherSuites []TLSCipherType `json:"cipherSuites"`
+}
+
+// EnvoyListener defines parameters for an Envoy Listener.
+type EnvoyListener struct {
+	// Defines an Envoy Listener Address.
+	// +kubebuilder:validation:MinLength=1
+	Address string `json:"address"`
+
+	// Defines an Envoy listener Port.
+	Port int `json:"port"`
+
+	// AccessLog defines where Envoy logs are outputted for this listener.
+	AccessLog string `json:"accessLog"`
+}
+
+// EnvoyLogging defines how Envoy's logs can be configured.
+type EnvoyLogging struct {
+	// AccessLogFormat sets the global access log format.
+	// Valid options are 'envoy' or 'json'
+	// +kubebuilder:default="envoy"
+	// +kubebuilder:validation:Enum="envoy";"json"
+	AccessLogFormat AccessLogType `json:"accessLogFormat"`
+
+	// AccessLogFormatString sets the access log format when format is set to `envoy`.
+	// When empty, Envoy's default format is used.
+	// +optional
+	AccessLogFormatString *string `json:"accessLogFormatString,omitempty"`
+
+	// AccessLogFields sets the fields that JSON logging will
+	// output when AccessLogFormat is json.
+	// +optional
+	AccessLogFields AccessLogFields `json:"jsonFields,omitempty"`
+}
+
+// TimeoutParameters holds various configurable proxy timeout values.
+type TimeoutParameters struct {
```

Comment:
I did figure out a validation for timeout strings that you could use here: https://github.com/projectcontour/contour/blob/main/apis/projectcontour/v1/httpproxy.go#L622

Perhaps worth defining a timeout string type alias too, to consolidate the validation instead of copy-pasting it everywhere, but we can do that separately.